### PR TITLE
[bitnami/kafka] Advertisedlisteners when external service and NodePort

### PR DESCRIPTION
**Pull request by sveba**
**Description of the change**

Using HOST_IP for "advertisedListeners"

**Benefits**

"advertisedListeners" with correct value 

**Possible drawbacks**

**Applicable issues**

Fixes EXTERNAL in "advertisedListeners"  when external service, autodiscovery and NodePort enabled. 
Until now it was PORT:PORT. With this fix it is IP:PORT. 

**Additional information**

**Checklist** 
- [X ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X ] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@ data:
     export EXTERNAL_ACCESS_PORT={{ .Values.externalAccess.service.port }}
     {{- else if eq .Values.externalAccess.service.type "NodePort" }}
     {{- if .Values.externalAccess.autoDiscovery.enabled }}
-    export EXTERNAL_ACCESS_IP="$(<${SHARED_FILE})"
+    export EXTERNAL_ACCESS_IP="${HOST_IP}"
```
Comment (juan131) [inline bitnami/kafka/templates/scripts-configmap.yaml:92]: I guess we don't need this block anymore in the auto-discovery.sh script:
```
    {{- else if eq .Values.externalAccess.service.type "NodePort" }}
    k8s_svc_node_port() {
        local namespace=${1:?namespace is missing}
        local service=${2:?service is missing}
        local index=${3:-0}
        local node_port="$(kubectl get svc "$service" -n "$namespace" -o jsonpath="{.spec.ports[$index].nodePort}")"
        echo "$node_port"
    }
    k8s_svc_node_port "{{ $releaseNamespace }}" "$SVC_NAME" | tee "$SHARED_FILE"
    {{- end }}
```

File: bitnami/kafka/templates/statefulset.yaml [hunk 2]
```diff
@@ -190,6 +190,12 @@ spec:
             {{- if .Values.externalAccess.autoDiscovery.enabled }}
             - name: SHARED_FILE
               value: "/shared/info.txt"
+            {{- if eq .Values.externalAccess.service.type "NodePort" }}
+            - name: HOST_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP 
+            {{- end }}
```
Comment (juan131) [inline bitnami/kafka/templates/statefulset.yaml:198]: Nice solution!

**Review by juan131 — COMMENTED** [review]
Thanks so much for this great contribution! Could you please check my comments?

File: bitnami/kafka/templates/scripts-configmap.yaml (hunk 1 above)
Comment (sveba) [reply bitnami/kafka/templates/scripts-configmap.yaml:92]: Actually it is still needed for resolving the PORTs here:
https://github.com/bitnami/charts/blob/457c7302ed0ecb660e77d307effe5cb050fb4fb7/bitnami/kafka/templates/scripts-configmap.yaml#L99

Comment (juan131) [reply bitnami/kafka/templates/scripts-configmap.yaml:92]: You're right!

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**